server/routes.go

```diff
@@ -339,6 +339,158 @@ func getDefaultSessionDuration() time.Duration {
 	return defaultSessionDuration
 }
 
+func (s *Server) EmbedHandler(c *gin.Context) {
```

Comment:
Possible to add a few handler tests for this?

Reply:
What would the handler tests look like / what would it test that isn't covered by the integration tests? routes_test.go seems to focus more on basic data-independent functionalities with a dummy model, not sure how embed handler would fit into that (similarly to how chat and generate have their own integration tests instead)